core/src/main/kotlin/org/jetbrains/research/testspark/core/generation/llm/LLMWithFeedbackCycle.kt

```diff
@@ -35,20 +44,36 @@ data class FeedbackResponse(
     init {
         if (executionResult == FeedbackCycleExecutionResult.OK && generatedTestSuite == null) {
             throw IllegalArgumentException("Test suite must be provided when FeedbackCycleExecutionResult is OK, got null")
-        } else if (executionResult != FeedbackCycleExecutionResult.OK && generatedTestSuite != null) {
```

Comment:
It will be logical to check that for `FeedbackCycleExecutionResult.NO_COMPILABLE_TEST_CASES_GENERATED` test suite is not null.

Reply:
Our `core` is also used by JB Academy team; their client code may not expect that `FeedbackCycleExecutionResult.NO_COMPILABLE_TEST_CASES_GENERATED` requires a test suite to be present.

Here, we merely ensure that once a client says that generation succeeded (by setting `FeedbackCycleExecutionResult.OK` as a result code), then the client is required to provide a valid test suite.

Reply:
Added the check for `FeedbackCycleExecutionResult.NO_COMPILABLE_TEST_CASES_GENERATED`.